sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy

```diff
@@ -315,21 +492,18 @@ class DirectoryAPITest extends APISpec {
         }
         def options = new DataLakePathCreateOptions().setMetadata(metadata)
 
-        when:
-        dc.createIfNotExistsWithResponse(options, null, Context.NONE)
-        def response = dc.getProperties()
-
-        then:
+        expect:
+        dc.createIfNotExistsWithResponse(options, null, null).getStatusCode() == statusCode
         // Directory adds a directory metadata value
         for(String k : metadata.keySet()) {
-            response.getMetadata().containsKey(k)
-            response.getMetadata().get(k) == metadata.get(k)
+            dc.getProperties().getMetadata().containsKey(k)
```

Comment:
Why overwrite the value of dc from the setup?

You've also introduced extra rest requests by calling getProperties an additional time.

Reply:
I generate new dc because if we use existing dc it stores the existing data associated with the directory client, and throws the following: 
Status code 409, "{"error":{"code":"PathAlreadyExists","message":"The specified path already exists.

Reply:
I generate new dc because if we use existing dc it stores the existing data associated with the directory client, and throws the following: 
Status code 409, "{"error":{"code":"PathAlreadyExists","message":"The specified path already exists.